addons/account/models/chart_template.py

```diff
@@ -136,6 +151,78 @@ class AccountChartTemplate(models.Model):
         help="Account that will be set on lines created in cash basis journal entry and used to keep track of the "
              "tax base amount.")
 
+    @api.constrains('currency_id', 'parent_id')
+    def _constraints_required_parent(self):
+        for record in self:
+            if not record.currency_id and not record.parent_id.currency_id:
+                raise ValidationError(_("No currency set on %s nor its parent"))
+            if not record.bank_account_code_prefix and not record.parent_id.bank_account_code_prefix:
+                raise ValidationError(_("bank account code prefix set on %s nor its parent"))
+            if not record.cash_account_code_prefix and not record.parent_id.cash_account_code_prefix:
+                raise ValidationError(_("cash account code prefix set on %s nor its parent"))
+            if not record.transfer_account_code_prefix and not record.parent_id.transfer_account_code_prefix:
+                raise ValidationError(_("transfer account code prefix set on %s nor its parent"))
```

Comment:
You could just check that if you don't have a parent_id, then all the fields must be set instead of having a recursive check